src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java

```diff
@@ -218,18 +220,11 @@ public void doWork() {
                 }
                 Map<String, Long> peerWaterMarks = peerWaterMarksByTerm.get(currTerm);
 
-                long quorumIndex = -1;
-                for (Long index : peerWaterMarks.values()) {
-                    int num = 0;
-                    for (Long another : peerWaterMarks.values()) {
-                        if (another >= index) {
-                            num++;
-                        }
-                    }
-                    if (memberState.isQuorum(num) && index > quorumIndex) {
-                        quorumIndex = index;
-                    }
-                }
+                List<Long> sortedWaterMarks = peerWaterMarks.values()
+                        .stream()
+                        .sorted((o1, o2) -> (int)(o2 - o1))
+                        .collect(Collectors.toList());
+                long quorumIndex = sortedWaterMarks.get(sortedWaterMarks.size() / 2);
```

Comment:
If the number of nodes is even, it should get `sortedWaterMarks.size() / 2 - 1` as quorumIndex.

Reply:
@RongtongJin 
[3, 2, 1]
quorumIndex = 2, elem at  `1 = len/2`
[4, 3, 2, 1]
quorumIndex = 2, elem at  `2 = len/2`

Reply:
You're right.